### ISSUE TITLE
Enhancing landmark training in imglab

### ISSUE THREAD

**Issue by Andyjk1406** [davisking/dlib#1145]
My understanding is that object bounding boxes need to be accurate for landmark training.

I want to train a fresh set of facial landmarks, but dlib already has good face detectors (especially the DNN one!) so I don't want to create bounding boxes manually in imglab, then label the landmarks.

To create the training data, is there:
a) A benefit to using the DNN to detect the face, then manually landmarking within the box (ie ensuring consistent object detection) ?
b) If so, is there any way to do this at the moment, or do I need to recode imglab to autodetect the bounding boxes?

I just want to check that what I am suggesting is sensible?

**Comment by davisking** [comment]
In all ML algorithms, the training data must match the testing data. So in this context it means that the bounding box the landmark training sees should be the same bounding box you would see during testing, which is the box from the face detector. So yes, you need to use boxes from face detection for training the landmark model. Weather or not they are "accurate" is not particularly relevant in this context.

So run the face detector to get bounding boxes and use them to build your dataset.

**Comment by Andyjk1406** [comment]
Thanks

Yes by 'accurate' bounding box I mean 'consistent under some metric' (such as we might expect a consistent result from any face detector - possibly inaccurate...but consistent).

So the dlib workflow would be to create the xml with bounding boxes programatically (using one of the face detectors) then load that into imglab and mark on the landmarks. 
Or maybe just put boxes on in imglab just so that I can now mark landmarks, then in the training code, find the face with a detector and hope (check) all my landmarks are within the detected box.
